core/net/ipv6/uip6.c

```diff
@@ -1854,8 +1855,10 @@ uip_process(uint8_t flag)
       if((UIP_TCP_BUF->flags & TCP_SYN)) {
         if((uip_connr->tcpstateflags & UIP_TS_MASK) == UIP_SYN_RCVD) {
           goto tcp_send_synack;
+#if UIP_ACTIVE_OPEN
         } else if((uip_connr->tcpstateflags & UIP_TS_MASK) == UIP_SYN_SENT) {
           goto tcp_send_syn;
```

Comment:
Based on the other `goto tcp_send_syn;` lines in this file, I think that there should be a `UIP_TCP_BUF->flags = 0;` line above this line.